File: backend/src/models/Ingredient.js
const mongoose = require('mongoose');

const ingredientSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  unit: { type: String, required: true }, // e.g. kg, grams, pcs, liters, ml
  currentStock: { type: Number, default: 0, min: 0 },
  lowStockThreshold: { type: Number, default: 10 },
  costPerUnit: { type: Number, default: 0 },
  branchId: { type: String, default: 'branch-1' }
}, { timestamps: true });

ingredientSchema.index({ name: 1, branchId: 1 }, { unique: true });

module.exports = mongoose.model('Ingredient', ingredientSchema);

File: frontend/lib/src/models/ingredient.dart
class Ingredient {
  final String id;
  final String name;
  final double currentStock;
  final String unit;
  final double costPerUnit;

  Ingredient({
    required this.id,
    required this.name,
    required this.currentStock,
    required this.unit,
    required this.costPerUnit,
  });

  factory Ingredient.fromMap(Map<String, Object?> map) {
    return Ingredient(
      id: map['id'] as String,
      name: map['name'] as String,
      currentStock: (map['current_stock'] as num).toDouble(),
      unit: map['unit'] as String,
      costPerUnit: (map['cost_per_unit'] as num).toDouble(),
    );
  }

  Map<String, Object?> toMap() {
    return {
      'id': id,
      'name': name,
      'current_stock': currentStock,
      'unit': unit,
      'cost_per_unit': costPerUnit,
    };
  }
}
